### PR TITLE
Bug fixes in SnippetClient.check_app_installed

### PR DESCRIPTION
**Pull request by xpconanfan**
Fix an issue where the function fails due to trailing `^M` in adb output.
Also add unit tests for the function.

### CODE REVIEW

File: mobly/controllers/android_device_lib/snippet_client.py [hunk 1]
```diff
@@ -78,25 +80,27 @@ def check_app_installed(self):
         """Overrides superclass."""
         # Check that the Mobly Snippet app is installed.
         if not self._adb_grep_wrapper(
-            'pm list package | grep ^package:%s$' % self.package):
+                'pm list package | grep "^package:%s[^M]\?$"' % self.package):
```
Comment (adorokhine) [inline mobly/controllers/android_device_lib/snippet_client.py]: [^M] means "not capital letter M":
$ echo "a" | grep "^a[^M]\?$"
a
$ echo "aM" | grep "^a[^M]\?$"
$ echo "aB" | grep "^a[^M]\?$"
aB

Comment (xpconanfan) [inline mobly/controllers/android_device_lib/snippet_client.py]: How did you type it? I used `Ctrl+V; Ctrl+M` in cli.
But not sure how python interprets it..

Comment (adorokhine) [inline mobly/controllers/android_device_lib/snippet_client.py]: It's literally a caret followed by a capital M. You can look at your source file in a hex editor. 5E 4D.

Comment (xpconanfan) [inline mobly/controllers/android_device_lib/snippet_client.py]: Yeah, checked, so i'm not sure how to write the real newline here...

File: tests/mobly/controllers/android_device_lib/snippet_client_test.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        self.apk_not_instrumented = kwargs.get('apk_not_instrumented', False)
+        self.target_not_installed = kwargs.get('target_not_installed', False)
+
+    def shell(self, params):
+        if 'pm list package' in params:
+            if self.apk_not_installed:
+                return b''
+            if self.target_not_installed and MOCK_MISSING_PACKAGE_NAME in params:
+                return b''
+            return bytes('package:%s\\r' % MOCK_PACKAGE_NAME, 'utf-8')
```
Comment (adorokhine) [inline tests/mobly/controllers/android_device_lib/snippet_client_test.py]: raw string here too (and elsewhere in this file)

**Pull request merged**